src/charm.py

```diff
@@ -9,250 +9,61 @@
-        # collect-status will log an error and set blocked if the config is invalid
-        cfg = self.config.get("role")
-        if cfg in ("all", None):
-            return TempoRole.all
-        return TempoRole(cfg)
-
-    @property
-    def tempo_role(self) -> Optional[TempoRole]:
-        """The role of this tempo worker, if a valid one is assigned."""
-        try:
-            return self._tempo_role
-        except ValueError:
-            return None
-
-    def _update_tls_certificates(self) -> bool:
-        # caller is responsible for guarding for container connectivity
-        if privkey := self.tempo_cluster.get_privkey():
-            ca_cert, server_cert = self.tempo_cluster.get_ca_and_server_certs()
-            self.tempo.update_certs(privkey, ca_cert, server_cert)
-            if ca_cert:
-                # update cacert in charm container too, for self-instrumentation
-                Path(TEMPO_CA_FILE).write_text(ca_cert)
-                subprocess.run(["update-ca-certificates", "--fresh"])
-            return True
-
-        else:
-            self.tempo.clear_certs()
-
-            # clear from charm container too
-            Path(TEMPO_CA_FILE).unlink(missing_ok=True)
-            subprocess.run(["update-ca-certificates", "--fresh"])
-            return True
-
-    def pre_update_checklist(
-        self,
-    ) -> List[str]:
-        """Check if the charm as a whole is ready, and if not, return a list of failed checks."""
-        failures = []
-        failures.extend(self.tempo.pre_update_checklist())
-
-        if not self.tempo_cluster.get_tempo_config():
-            failures.append("tempo_cluster config not ready")
-
-        return failures
-
     def _on_collect_status(self, e: CollectStatusEvent):
-        if not self._container.can_connect():
-            e.add_status(WaitingStatus("Waiting for `tempo` container"))
-        if not self.model.get_relation("tempo-cluster"):
-            e.add_status(
-                BlockedStatus("Missing tempo-cluster relation to a tempo-coordinator charm")
-            )
-        elif not self.tempo_cluster.relation:
-            e.add_status(WaitingStatus("Tempo-Cluster relation not ready"))
-        if not self.tempo_cluster.get_tempo_config():
-            e.add_status(WaitingStatus("Waiting for coordinator to publish a tempo config"))
-
-        if role := self.tempo_role:
-            e.add_status(
-                ActiveStatus(
-                    "(all roles) ready." if role is TempoRole.all else f"{role.name} ready."
-                )
-            )
-        else:
+        self.worker._on_collect_status(e)
+        # add Tempo worker custom blocking conditions
+        if self.worker.roles and self.worker.roles[0] not in self._valid_roles:
             logger.error(
                 f"`role` config value {self.config.get('role')!r} invalid: should "
-                f"be one of {list(TempoRole)}."
+                f"be one of {(self._valid_roles)}."
```

Comment:
This will be logged every hook. Is that desirable?

Reply:
following the pattern in shared worker's `collect_unit_status`, maybe we don't actually. Setting the status would be enough I guess.